OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -121,7 +121,7 @@ public ProductionQueue(ActorInitializer init, Actor playerActor, ProductionQueue
 			self = init.Self;
 			Info = info;
 			playerResources = playerActor.Trait<PlayerResources>();
-			playerPower = playerActor.Trait<PowerManager>();
+			playerPower = playerActor.TraitOrDefault<PowerManager>();
```

Comment:
You'll need to move this to INotifyCreated.Created(). Without Requires<> enforcing the prodqueue be only loaded after the powermanager, it could be that the queue ends up without initializing the power trait within itself.